etl/src/v2/state/store/base.rs

```diff
@@ -1,46 +1,48 @@
-use postgres::schema::Oid;
-use std::future::Future;
+use postgres::schema::TableId;
+use std::collections::HashMap;
 use thiserror::Error;
 
-use crate::v2::pipeline::PipelineId;
-use crate::v2::state::origin::ReplicationOriginState;
-use crate::v2::state::table::TableReplicationState;
+use crate::v2::{
+    replication::slot::SlotError,
+    state::{store::postgres::ToTableStateError, table::TableReplicationPhase},
+};
 
 #[derive(Debug, Error)]
 pub enum StateStoreError {
-    #[error("Replication origin state not found in store")]
-    ReplicationOriginStateNotFound,
+    #[error("Sqlx error in state store: {0}")]
+    Sqlx(#[from] sqlx::Error),
 
-    #[error("Table replication state not found in store")]
-    TableReplicationStateNotFound,
-}
+    #[error("Slot error in state store: {0}")]
+    Slot(#[from] SlotError),
 
-pub trait StateStore {
-    fn load_replication_origin_state(
-        &self,
-        pipeline_id: PipelineId,
-        table_id: Option<Oid>,
-    ) -> impl Future<Output = Result<Option<ReplicationOriginState>, StateStoreError>> + Send;
+    #[error("Invalid confirmed flush lsn value in state store: {0}")]
+    InvalidConfirmedFlushLsn(String),
+
+    #[error("Missing slot in state store: {0}")]
+    MissingSlot(String),
+
+    #[error("Error converting from table replication phase to table state")]
+    ToTableState(#[from] ToTableStateError),
+}
 
-    fn store_replication_origin_state(
+#[async_trait::async_trait]
```

Comment:
Added to make the trait dyn compatible. We might consider to go the static dispatch route if this becomes a perf bottleneck.